public/app/plugins/datasource/loki/responseUtils.ts

```diff
@@ -27,3 +27,14 @@ export function extractLogParserFromDataFrame(frame: DataFrame): { hasLogfmt: bo
 
   return { hasLogfmt, hasJSON };
 }
+
+export function extractHasErrorLabelFromDataFrame(frame: DataFrame): boolean {
+  const labelField = frame.fields.find((field) => field.name === 'labels' && field.type === FieldType.other);
+  if (labelField == null) {
+    return false;
+  }
+
+  const labels: Array<{ [key: string]: string }> = labelField.values.toArray();
+  const hasErrorLabel = labels.find((label) => label['__error__']);
```

Comment:
You could use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some here.

Reply:
yes we can! good suggestion.